src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -1,186 +1,346 @@
+					var date = DateTimeOffset.FromUnixTimeMilliseconds(location.Time);
+					if (date + maximumAge > startDate)
+					{   // can be used, but is it best accuracy?
+						if (location.HasAccuracy)
+						{
+							if (location.Accuracy < bestAccuracy)
+							{
+								bestAccuracy = (int)location.Accuracy;
+								bestLocation = location;
+							}
+						}
+						else
+							bestLocation = location;
+					}
+				}
+			}
+
+			if (bestLocation != null)
+			{
+				RemoveUpdates();
+				BroadcastStatus(PositionStatus.Ready);
+				return bestLocation.ToGeoPosition();
+			}
+
+			// wait for fix
+			for (int i = (int)(timeout.TotalMilliseconds / 250.0); i > 0; i--)
+			{
+				await Task.Delay(250);
+				if (_locChanged)
+				{
+					RemoveUpdates();
+					BroadcastStatus(PositionStatus.Ready);
+					return _location.ToGeoPosition();
+				}
+			}
+
+			BroadcastStatus(PositionStatus.Disabled);
+			RemoveUpdates();
+			throw new TimeoutException("Timeout in GetGeopositionAsync(TimeSpan,TimeSpan)");
+
+		}
+
+		~Geolocator()
+        {
+			RemoveUpdates();
+			_locationManager?.Dispose();
+        }
+
+		private void RemoveUpdates()
+		{
+			if (this is null)
+			{ // when caled from destructor
+				return;
+			}
+			_locationManager?.RemoveUpdates(this);
+		}
+
+		private void RequestUpdates()
+		{
+			if (_desiredAccuracyInMeters.HasValue)
+			{
+				if (_desiredAccuracyInMeters.Value < 100)
+				{
+					_locationCriteria.HorizontalAccuracy = Accuracy.High;
+				}
+				else
+					if (_desiredAccuracyInMeters.Value < 500)
+				{
+					_locationCriteria.HorizontalAccuracy = Accuracy.Medium;
+				}
+				else
+				{
+					_locationCriteria.HorizontalAccuracy = Accuracy.Low;
+				}
+
+			}
+			else
+			{
+				_locationCriteria.HorizontalAccuracy = Accuracy.Medium;
+			}
+
+			_locationManager?.RequestLocationUpdates(_reportInterval, (float)_movementThreshold, _locationCriteria, this, Looper.MainLooper);
+		}
+
+		private uint _reportInterval = 1000;
+		private double _movementThreshold = 0;
+
+		static bool _locChanged = false;
+		static Location _location;
+
+		private Criteria _locationCriteria = new Android.Locations.Criteria() { HorizontalAccuracy = Accuracy.Medium };
+
+
+		public uint ReportInterval
+		{
+			get => _reportInterval;
+			set
+			{
+				_reportInterval = value;
+				RemoveUpdates();
+				RequestUpdates();
+			}
+		}
+
+		public double MovementThreshold
+		{
+			get => _movementThreshold;
+			set
+			{
+				_movementThreshold = value;
+				RemoveUpdates();
+				RequestUpdates();
+			}
+		}
+
+		public uint? DesiredAccuracyInMeters
+		{
+			get => _desiredAccuracyInMeters;
+			set
+			{
+				_desiredAccuracyInMeters = value;
+				RemoveUpdates();
+				RequestUpdates();
+			}
+		}
+
+#region Android ILocationListener
+		public void OnLocationChanged(Location location)
+		{
+			DateTimeOffset date = DateTimeOffset.FromUnixTimeMilliseconds(location.Time);
+			if (date.AddMinutes(1) > DateTimeOffset.UtcNow)

```

Comment:
Yes, this is arbitrary.
Android, asked for location, reports last known location, even days old. So I need to filter out obsolete information; what is obsolete is hard to know. We can make some calculations, based on timestamp of last location and speed, but it would be arbitrary anyway (besides, not always speed is known). And much simpler is test for location age. Why one minute? Because it seems appropriate TimeSpan for me :)

Reply:
If you're making an arbitrary decision, let it be known in the code, and make it a constant that is easy to find at the top.